eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -359,6 +359,25 @@ jobs:
     displayName: Validate Characters are ASCII
     condition: eq(variables['AZ_SDK_CODE_COV'], 1)
 
+  # Validate all the files are formatted correctly according to the .clang-format file.
+  - bash: |
+      # Run clang-format recursively on each source and header file within the repo.
+      find . \( -iname '*.h' -o -iname '*.c' \) -exec clang-format -i {} \;
+      
+      git status --untracked-files=no --porcelain
+      
+      if [[ `git status --untracked-files=no --porcelain` ]]; then
+        echo Some files were not formatted correctly according to the .clang-format file.
+        echo Please run clang-format to fix the issue.
+        exit 1
+      fi
+      
+      echo Success, all files are formatted correctly according to the .clang-format file.
+      exit 0
+      
+    displayName: Validate Clang Format
+    condition: eq(variables['AZ_SDK_CODE_COV'], 1)
```

Comment:
Nit, Its a little odd we condition this and the utf bom check to run only if a code coverage build.

Reply:
I agree. It's a simple way to run the checks on linux, but only on one leg/once. That leg takes < 5 minutes to run, and is one of the quickest, which makes it a good candidate. It's a stop gap.